packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -386,20 +372,10 @@ interface Message {
 	readonly changeset: JsonCompatibleReadOnly;
 }
 
-/**
- * Observes Changesets (after rebase), after writes data into summaries when requested.
- */
-export interface Index {
-	/**
-	 * If provided, records data into summaries.
-	 */
-	readonly summaryElement?: SummaryElement;
-}
-
 /**
  * Specifies the behavior of an {@link Index} that puts data in a summary.
  */
-export interface SummaryElement {
+export interface IndexSummarizer {
```

Comment:
The usage of the term `Index` (both in the doc comment and the `IndexSummarizer` name) is awkward without the above interface that defines what an index is.
How about just `Summarizer` or `SummaryContributor`?

Reply:
It seems that there are other usages of "index" that may need updating (e.g. `MockIndexSummarizer`).

Reply:
Good call, I did several renames. I don't want to purge the term "index" entirely from comments and such, because we _are_ still calling the forest, schema, identifier map, etc. "indexes", but you're right that it probably shouldn't show up right now in an actual type symbol.